engine/crates/engine/registry-upgrade/src/lib.rs

```diff
@@ -444,24 +452,30 @@ fn insert_validators(
 
 fn convert_meta_field_type(
     ty: registry_v1::MetaFieldType,
-    type_ids: &HashMap<String, registry_v2::ids::MetaTypeId>,
+    type_ids: &HashMap<String, MetaTypeId>,
 ) -> MetaFieldTypeRecord {
     MetaFieldTypeRecord {
         wrappers: wrappers_from_string(ty.as_str()),
-        target: type_ids[ty.base_type_name()],
+        target: lookup_type_id(type_ids, ty.base_type_name()),
     }
 }
 
 fn convert_input_value_type(
     ty: registry_v1::InputValueType,
-    type_ids: &HashMap<String, registry_v2::ids::MetaTypeId>,
+    type_ids: &HashMap<String, MetaTypeId>,
 ) -> MetaInputValueTypeRecord {
     MetaInputValueTypeRecord {
         wrappers: wrappers_from_string(ty.as_str()),
-        target: type_ids[ty.base_type_name()],
+        target: lookup_type_id(type_ids, ty.base_type_name()),
     }
 }
 
+fn lookup_type_id(type_ids: &HashMap<String, MetaTypeId>, name: &str) -> MetaTypeId {
```

Comment:
for a better error?

Reply:
Yeah, a bug was causing the lookup to fail and "no thing at index" is the most useless error message.